implementation/src/main/java/io/smallrye/config/Converters.java

```diff
@@ -130,6 +130,10 @@ private Converters() {
         return null;
     });
 
+    @SuppressWarnings("unchecked")
+    static final Converter<Short> SHORT_CONVERTER = BuiltInConverter.of(12,
```

Comment:
You need to add the SHORT_CONVERTER for the `12` value in the `readResolve`method  https://github.com/smallrye/smallrye-config/pull/180/files#diff-dd5791f68a75fe49b471b3733ecd25aeR601

Reply:
Thank you for the review @jmesnil :) I've updated the PR to include that change.